lib/Interface/GraphContainer.cpp

```diff
@@ -0,0 +1,536 @@
+                            std::cout << x << "  ";
+                          }
+                          break;
+                        case graph::detail::
+                            GRAPH_INC_MATRIX_UNDIRECTED_UNWEIGHTED:
+                          for (T x : this->incMat[i]) {
+                            std::cout << x << " ";
+                          }
+                          break;
+                        case graph::detail::
+                            GRAPH_INC_MATRIX_UNDIRECTED_WEIGHTED:
+                          for (T x : this->incMat[i]) {
+                            std::cout << x << " ";
+                          }
+                          break;
+                        case graph::detail::GRAPH_INC_MATRIX_DIRECTED_WEIGHTED:
+                          for (T x : this->incMat[i]) {
+                            std::cout << x << " ";
+                          }
+                          break;
+                        case graph::detail::GRAPH_ADJ_MATRIX_DIRECTED_WEIGHTED:
+                          for (T x : this->adjMat[i]) {
+                            std::cout << x << " ";
+                          }
+                          break;
+                        case graph::detail::
+                            GRAPH_ADJ_MATRIX_UNDIRECTED_WEIGHTED:
+                          for (T x : this->adjMat[i]) {
+                            std::cout << x << " ";
+                          }
+                          break;
+                        case graph::detail::
+                            GRAPH_ADJ_MATRIX_DIRECTED_UNWEIGHTED:
+                          for (T x : this->adjMat[i]) {
+                            std::cout << x << " ";
+                          }
+                          break;
+                        case graph::detail::
+                            GRAPH_ADJ_MATRIX_UNDIRECTED_UNWEIGHTED:
+                          for (T x : this->adjMat[i]) {
+                            std::cout << x << " ";
+                          }
+                          break;
+        }
+        std::cout << std::endl;
+    }
+};
+/**
+ * @brief Function to print the graph using the memref descriptor
+ * 
+ * @tparam T Repesents the datatype used  
+ * @tparam N represents the number of dimensions.
+ */
+template <typename T, size_t N> void Graph<T, N>::printGraph() {
+  for (int v = 0; v < this->sizes[0]; ++v) {
+    for (int w = 0; w < this->sizes[1]; ++w) {
+      std::cout << this->aligned[this->sizes[0] * v + w] << " ";
+    }
+    std::cout << std::endl;
+  }
+}
+/**
+ * @brief Converts the various Graph representations types to a linear memref descriptor
+ * 
+ * @tparam T represents the datatype used
+ * @tparam N represents the number of dimensions.
+ * @return MemRef_descriptor returns a memref descriptor of the respective graph representation type after conversion.
+ */
+
+template <typename T, size_t N> 
+MemRef_descriptor Graph<T, N>::graph_to_MemRef_descriptor()
```

Comment:
Didn't understood the logic behind using the `MemRef_descriptor` when you have already created your custom `container.h`, and also creating the graph class by inheriting it.

Reply:
Resolved and loaded all required data into an object of the MemRef class in this [commit](https://github.com/meshtag/GraphMLIR/pull/19/commits/5ca64fe5dc76d07f58460c41c64c019a5baae26c). Please have a look.